### PR TITLE
add blueprint `to_json` method for standalone rendering

### PR DESCRIPTION
**Pull request by troyready**
Enables blueprint files to use a few simple lines to output their template:
```
if __name__ == "__main__":
    from stacker.context import Context
    print BPClassName('test', Context({"namespace": "test"})).to_json()
```

### CODE REVIEW

File: stacker/blueprints/base.py [hunk 1]
```diff
@@ -462,6 +463,16 @@ def render_template(self):
         version = hashlib.md5(rendered).hexdigest()[:8]
         return (version, rendered)
 
+    def to_json(self):
+        """Render the blueprint and return the template in json form."""
+
+        variables = []
+        for k in self.get_parameter_definitions():
+            variables.append(Variable(k, 'unused_value'))
```
Comment (phobologic) [inline stacker/blueprints/base.py]: Doesn't this break in the case of the variable not accepting a string type?

Comment (phobologic) [inline stacker/blueprints/base.py]: Maybe we could do something a little more intelligent here to cover the most basic cases - the basic python types are probably the easiest.

**Comment by troyready** [comment]
@phobologic All object types should be fine as long as they have a default value assigned to them; it's only CFN parameters in my experience that trip this up because they don't need really need default values for the template to be generated.

I'm not sure if it's too lazy to say that non-CFN parameters will require a default to use this method? Or maybe the method should extended to take something like a list Variable objects as a parameter to be passed to `resolve_variables`?

**Comment by phobologic** [comment]
@troyready Oh, I like the idea of this taking a dictionary of `variable type -> variable value`.  We could even provide a default set for most python types.  Another option is to have it call the value type directly, since they all should provide a default value:


```
>>> str()
''
>>> int()
0
>>> dict()
{}
>>> list()
[]
```

Mix that with the ability to provide a map for default values, and I think we have something pretty awesome here.  This would make the `dump` command possible again as well possibly :)

File: stacker/blueprints/base.py (hunk 1 above)
Comment (troyready) [reply stacker/blueprints/base.py]: `get_parameter_definitions` exclusively returns CFN parameters, so they will always take a string.

**Comment by troyready** [comment]
@phobologic I've finally gotten around to updating this and think it's in a good place now.

I've added the override dictionary, but couldn't find a simple way to set default value types. There's probably a simple method I'm missing, but I don't think it unreasonable to leave out anyway -- requiring that defaults be specified in the VARIABLES dictionary itself is nice for keeping things explicit.

**Pull request merged**